### PR TITLE
Create obs_base and make obs_test work with it Tickets/dm 7578

### PR DESCRIPTION
**Pull request by parejkoj**

### CODE REVIEW

File: tests/test_obs_test.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+        self.dataIds = {'raw': {'visit': 1, 'filter': 'g'},
+                        'bias': {'visit': 1},
+                        'flat': {'visit': 1},
+                        '1': {'visit': 1, 'filter': 'g'},
+                        '2': {'visit': 2, 'filter': 'g'},
+                        '3': {'visit': 3, 'filter': 'r'}
+                        }
+        self.path_to_raw = os.path.join(self.data_dir, "raw", "raw_v1_fg.fits.gz")
+        self.result_ids = {'raw': 1, 'bias': 1, 'flat': 1}
```
Comment (r-owen) [inline tests/test_obs_test.py]: I find this name confusing; what kind of IDs does this contain? If it's a visit ID, then `result_visits` would probably be clearer

File: tests/test_obs_test.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import lsst.afw.geom
+import lsst.obs.base.tests
+import lsst.obs.test
+
+
+class TestObsTest(lsst.obs.base.tests.ObsTests, unittest.TestCase):
+    def setUp(self):
+        product_dir = getPackageDir('obs_test')
+        self.data_dir = os.path.join(product_dir, 'data', 'input')
+        self.Mapper = lsst.obs.test.TestMapper
```
Comment (r-owen) [inline tests/test_obs_test.py]: I wonder if you will find you want the class or an instance. If the class, please consider appending Class to the instance variable name, for clarity (self.MapperClass is clearly not an instance of a Mapper)

Comment (parejkoj) [inline tests/test_obs_test.py]: I've redone this to be a an instance, instantiated in the child class.

File: tests/test_obs_test.py (hunk 1 above)
Comment (parejkoj) [inline tests/test_obs_test.py]: They are exposureIds. I've renamed it. Thanks.

File: tests/test_obs_test.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+class TestObsTest(lsst.obs.base.tests.ObsTests, lsst.utils.tests.TestCase):
+    def setUp(self):
+        product_dir = getPackageDir('obs_test')
+        self.data_dir = os.path.join(product_dir, 'data', 'input')
+
+        super(TestObsTest, self).setUp()
+
+        self.butler = lsst.daf.persistence.Butler(root=self.data_dir)
+        self.mapper = lsst.obs.test.TestMapper(root=self.data_dir)
```
Comment (PaulPrice) [inline tests/test_obs_test.py]: This is breaking encapsulation.  `TestObsTest` should only set things that it cares about; it should not have to set up things for its parent class.  If something needs to be set up for the parent class, then you should call a method in the parent class to do that.  I should be able to change the names of instance variables in the parent class without this subclass knowing anything about it (I know that's not always possible, but it's impossible to such a large extent here that I think it's a bad idea).

**Pull request merged**